### PR TITLE
Add support for properly theming the property page controls

### PR DESCRIPTION
**Pull request by tannergooding**
This updates the ApplicationDesigner to support "themed" property pages. This, or something similar, will be required to ensure we meet the accessibility requirements around color contrast ratios.

A relevant comment that was in "PropPageDesignerView" was:
> 'We used to use a lighter color than SystemColors.Control (and we would get it from the color service).
>            '  But that causes issues because a) we can't control disabled control colors, b) we have problems when our
>            '  pages are hosted in another property page frame with different colors (e.g., C++), and c) we have problems
>            '  when hosting a native property page over which we have no color control.
>            'So now we simply use SystemColors.Control all the time (constant on PropPageUserControlBase).
>            'Me.BackColor = GetPropertyPageBackColor(System.Drawing.SystemColors.ControlLight)

The side effect being that hosted controls which are not "themed" will result in the pages being "unviewable" under the dark (or high contrast) themes. I worked around this issue by exposing a new property on `PropPageBase` and `PropPageUserControlBase` called `SupportsTheming`. This property is overridable and defaults to `false`. If the property returns `false`, then we continue with the previous logic (just using `SystemColors.Control`); otherwise, we pull our colors from the theming service.

This results in non-themed controls appearing as follows:
image

A future PR will update our property pages to return `SupportsTheming=True` and to pull the colors from the theming service.

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @dotnet/project-system

**Comment by tannergooding** [comment]
Also, FYI. @dpoeschl 

This sets us up for resolving DevDiv Bug 411550 (Luminosity contrast ratio for the Selected  tab item in does not match 4.5:1), which will require a separate update to ShellColors.xml in VS.

**Comment by davkean** [comment]
Can you show a me a picture of the overflow button?

File: src/Microsoft.VisualStudio.AppDesigner/PropPageDesigner/PropPageDesignerView.vb [hunk 1]
```diff
@@ -512,6 +551,32 @@ Namespace Microsoft.VisualStudio.Editors.PropPageDesigner
...
+            End If
+
+            ConfigurationPanel.BackColor = BackColor
+            PropertyPagePanel.BackColor = BackColor
+        End Sub
+
+        ''' <summary>
+        ''' We've gotta tell the renderer whenver the system colors change...
+        ''' </summary>
+        ''' <param name="msg"></param>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.AppDesigner/PropPageDesigner/PropPageDesignerView.vb]: Can you remove these empty xml comments.

**Comment by davkean** [comment]
The code itself looks good.

**Comment by tannergooding** [comment]
Here are the pictures for all four themes and for both "unthemed" and "themed" property pages. 

The "themed" versions were just a quick hack around that particular property page to get things working (so some things, like hover, might not be hooked up yet). It also means that things like the border colors were not configured yet (they are controlled via `FlatAppearance.BorderColor`). An example of a fully themed button is:
image

### Unthemed Control

#### Blue Theme
image

#### Dark Theme
image

#### Light Theme
image

#### High Contrast Mode
image
image

### Themed Control

#### Blue Theme
image

#### Dark Theme
image

#### Light Theme
image

#### High Contrast Mode
image
image

**Comment by srivatsn** [comment]
Tagging @MattGertz

**Comment by tannergooding** [comment]
DevDiv PR 68364 covers the VS side changes.

**Pull request merged**